pkg/kbld/image/err.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package image
+
+type ErrImage struct {
+	err error
+}
+
+var _ Image = ErrImage{}
```

Comment:
Dig the "Null Object" pattern variant: "Error Object". ⭐ 

I would have likely included an `error` in the `BuiltImage`, myself.

But this is a great example of how to keep the overall system simpler by cordoning off a non-domain case (i.e. where a given resolution is destined to fail, independent of type: an "Erroring Image") into its own type (i.e. `ErrImage`).

One can completely understand `BuiltImage` without having to consider this case at all... it sits _just_ outside the sphere of the domain.

Reply:
it's not a common pattern i use, but seems like it did ok job here.